src/data/set/n_ary.lean

```diff
@@ -64,6 +63,24 @@ lemma forall_image2_iff {p : γ → Prop} :
   image2 f s t ⊆ u ↔ ∀ (x ∈ s) (y ∈ t), f x y ∈ u :=
 forall_image2_iff
 
+variables (f)
+
+@[simp] lemma image_prod : (λ x : α × β, f x.1 x.2) '' s ×ˢ t = image2 f s t :=
```

Comment:
Should we have a
```suggestion
@[simp] lemma image_prod : f '' s ×ˢ t = image2 (curry f) s t :=
```
lemma as well / instead? The LHS is more general here than in your lemma.

Reply:
Oh, it's just a move, so out of scope

Reply:
`image2_curry` is basically this already.